todo_app/app.py

```diff
@@ -61,52 +65,35 @@ def sort_item():
     @app.route('/mark', methods=['POST'])
     def alter_item():
         newState=str(request.form.get('item_state'))
-        itemId=str(request.form.get('item_id'))
+        itemId=request.form.get('item_id')
 
-        #get valid states/ lists for cards to be in
-        trelloLists= {v: k for k, v in getTrelloLists(trelloBoardID, key, token).items()}
-        if newState in trelloLists.keys():
-            updateTrelloCard(itemId, {"idList":trelloLists[newState]}, "PUT", key, token)
+        updQuery={"_id":ObjectId(itemId)},{"$set":{"idList":newState}}
+        updateTrelloCard(updQuery,"PUT")
+        
         return redirect("/")
 
     # Used to delete an item
     @app.route('/rem', methods=['POST'])
     def remove_item():
-        card_id=str(request.form.get('item_id'))
-        updateTrelloCard(card_id, {}, "DELETE", key, token)
+        card_id=ObjectId(request.form.get('item_id'))
+        updateTrelloCard({"_id":card_id},"DELETE")
         return redirect("/")
 
-    #Funtions to work with Trello cards
-    # Used to update an item
-    def updateTrelloCard (cardId, newState, method, key, token):
-        url='https://api.trello.com/1/cards/'+cardId+'?key='+key+'&token='+token
-        res=requestJson(method,url, newState)
-        return res
+    #Funtions to work with cards       
+    # Used to get the item cards from DB
+    def getTrelloCards():
+        cards=[]
+        for todoCard in todoCards.find():
+            cards.append(ToDoItem(todoCard))
 
-        
-    # Used to get the item cards from a given Trello board
-    def getTrelloCards(boardId, key, token):
-        cards=[]      
-        trelloLists=getTrelloLists(boardId, key, token)    
-        cardDict=getTrelloItems(boardId, 'cards', key, token)
-            
-        for card in cardDict:
-            cards.append(ToDoItem(trelloLists, card))
         return cards
 
-    # Used to get the board lists/states from a given Trello board
-    def getTrelloLists(boardId, key, token):
-        listDict=getTrelloItems(boardId, 'lists', key, token)
-        listMap={}
-        for list in listDict:
-            listMap[list["id"]]=list["name"]
-        return listMap
 
-    # used to get details regarding a boards cards/lists
-    def getTrelloItems(boardId, item, key, token):
-        url='https://api.trello.com/1/boards/'+boardId+'/'+item+'?key='+key+'&token='+token
-        res=requestJson("GET",url, {})
+    # Used to update an item
+    def updateTrelloCard (newState, method):
+        try: res=actionMap[method](newState)
+        except: res=actionMap[method](newState[0],newState[1])
+        
```

Comment:
This is a little confusing and likely to lead to bugs.

Is it really neater to wrap `todoCards.update_one`/`todoCards.delete_one` in a single `updateTrelloCard` function when they are such different functions? Before you were wrapping making HTTP requests to an API. So the interface is always HTTP and there's some common behaviour like key, token, parsing the body as JSON. That made sense.

But this function obscures what's happening and is also less convenient to use.
If you just make them separate functions you could have functions that are used like this `deleteTrelloCard(card_id)` instead of `updateTrelloCard({"_id":card_id},"DELETE")`.